bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetLightingHandler.java

```diff
@@ -194,163 +215,150 @@ protected String ownIdPrefix() {
-                            getThing().getUID());
-                    return;
+            logger.debug("  $BRI brightnessWhat {} --> {}", brightnessWhat, newBrightnessWhat);
+            if (brightnessWhat != newBrightnessWhat) {
+                int newBrightness = Lighting.levelToPercent(newBrightnessWhat);
+                updateState(CHANNEL_BRIGHTNESS, new PercentType(newBrightness));
+                if (msg.isOff()) {
+                    brightnessBeforeOff = brightness;
                 }
+                brightness = newBrightness;
+                logger.debug("  $BRI brightness CHANGED to {}", brightness);
+            } else {
+                logger.debug("  $BRI no change");
             }
         }
-        logger.debug("  $BRI latestBriWhat={} latestBriBeforeOff={} brightnessLevelRequested={}", latestBrightnessWhat,
-                latestBrightnessWhatBeforeOff, brightnessLevelRequested);
     }
 
     /**
-     * Returns a WHERE address string based on bridge type and unit (optional)
+     * Updates light on/off state based on a OWN Lighting event message received
      *
-     * @param unit the device unit
-     **/
-    @Nullable
-    protected String toWhere(String unit) {
-        Where w = deviceWhere;
-        if (w != null) {
-            OpenWebNetBridgeHandler brH = bridgeHandler;
-            if (brH != null && brH.isBusGateway()) {
-                return w.value();
+     * @param msg the Lighting message received
+     */
+    private void updateOnOffState(Lighting msg) {
+        OpenWebNetBridgeHandler brH = bridgeHandler;
+        if (brH != null) {
+            if (msg.isOn() || msg.isOff()) {
+                String channelId;
+                int switchId = 0;
+                if (brH.isBusGateway()) {
+                    channelId = CHANNEL_SWITCH;
+                } else {
+                    WhereZigBee w = (WhereZigBee) (msg.getWhere());
+                    if (WhereZigBee.UNIT_02.equals(w.getUnit())) {
+                        channelId = CHANNEL_SWITCH_02;
+                        switchId = 1;
+                    } else {
+                        channelId = CHANNEL_SWITCH_01;
+                    }
+                }
+                int currentSt = sw[switchId];
+                int newSt = (msg.isOn() ? 1 : 0);
+                if (newSt != currentSt) {
+                    updateState(channelId, (newSt == 1 ? OnOffType.ON : OnOffType.OFF));
+                    sw[switchId] = newSt;
+                    logger.debug("  {} ONOFF CHANGED to {}", ownId, newSt);
+                } else {
+                    logger.debug("  {} ONOFF no change", ownId);
+                }
             } else {
-                return w + unit;
+                logger.debug("updateOnOffState() Ignoring unsupported WHAT for thing {}. Frame={}", getThing().getUID(),
+                        msg.getFrameValue());
+                return;
             }
-        } else {
-            return null;
+
```

Comment:
This empty line introduces a new checkstyle warning. Can you also check the other two checkstyle warnings?

Reply:
Added new commit with checkstyle warnings solved (Jenkins build failed but not because of this binding